core/src/main/java/org/elasticsearch/action/update/UpdateResponse.java

```diff
@@ -109,4 +112,42 @@ public String toString() {
         builder.append(",shards=").append(getShardInfo());
         return builder.append("]").toString();
     }
+
+    private static final ConstructingObjectParser<UpdateResponse, Void> PARSER;
+    static {
+        PARSER = new ConstructingObjectParser<>(UpdateResponse.class.getName(),
+                args -> {
+                    // index uuid and shard id are unknown and can't be parsed back for now.
+                    String index = (String) args[0];
+                    ShardId shardId = new ShardId(new Index(index, IndexMetaData.INDEX_UUID_NA_VALUE), -1);
+                    String type = (String) args[1];
+                    String id = (String) args[2];
+                    long version = (long) args[3];
+                    ShardInfo shardInfo = (ShardInfo) args[5];
+                    Long seqNo = (Long) args[6];
+
+                    Result result = null;
+                    for (Result r : Result.values()) {
+                        if (r.getLowercase().equals(args[4])) {
+                            result = r;
+                            break;
+                        }
+                    }
+
+                    UpdateResponse updateResponse = null;
+                    if (shardInfo != null && seqNo != null) {
+                        updateResponse = new UpdateResponse(shardInfo, shardId, type, id, seqNo, version, result);
+                    } else {
+                        updateResponse = new UpdateResponse(shardId, type, id, version, result);
+                    }
+                    return updateResponse;
+                });
+
+        DocWriteResponse.declareParserFields(PARSER);
+        PARSER.declareObject(UpdateResponse::setGetResult, (p, c) -> GetResult.fromXContentEmbedded(p), new ParseField(GET));
```

Comment:
You can do sth. like this I think instead of directly setting the parsed (incomplete GetResult): 
```
(x, y) -> x.setGetResult(new GetResult(x.getIndex(), x.getType(), x.getId(), y.getVersion(), y.isExists(), y.internalSourceRef(), y.getFields()))
```
This way you can use the index/type/id from the parent UpdateResponse together with the rest of the parsed GetResult information. I haven't tested this very thoroughly though because didn't completely understand the test setup yet.